src/client/tile.h

```diff
@@ -284,7 +283,7 @@ struct TileLayer
 	//! If true, the tile has its own color.
 	bool has_color = false;
 
-	std::shared_ptr<std::vector<FrameSpec>> frames = nullptr;
```

Comment:
Actually, std::unique_ptr is a good substitude for what you are trying to achieve since it implements the concept of pointer ownership while avoiding the locking mechanisms used by std::shared_ptr.

Reply:
Also, since std::vector encapsulates a pointer itself, why not just use it as is?